### ISSUE TITLE
Sometimes priority of song in playlist cannot be set (priority list is grayed out)

### ISSUE THREAD

**Issue by xhpohanka** [CDrummond/cantata#1054]
I do not know exactly what causes this issue but often I cannot set priority of song in playlist, all items are grayed out in Cantata context menu. It happens to me when using big playlists (~8000 songs).

Setting priority in ncmpcpp and other clients works without any problems.

**Comment by CDrummond** [comment]
Sorry, but I cannot reproduce this. You say all items are grey-ed out - I assume you mean on the main play queue context menu? Or only in the priority sub menu?

Which version of Cantata? Which OS?

I'm going to need more info. If it happens often, what are you doing before? How many items do you have selected? Is the play queue search field open? If so, any text entered?

**Comment by xhpohanka** [comment]
Arch Linux up to date, Cantata 2.0.1, mpd 0.20.9 running on another pc also with Arch Linux.
Please check this screenshot (sorry for Czech language, but I think you see what I'm talking about)

![cantata](https://user-images.githubusercontent.com/2233252/28994931-3b87b53a-79db-11e7-821f-9e56ffe9df87.png)

**Comment by CDrummond** [comment]
Sorry for the delay in replying - I was on holiday :-)

Can you pleas try with 2.1.0 - I see you are using 2.0.1 which is quite old. 2.1.0 was release just over a week ago.

What is odd is that only the 'Set priority' action itself should be enabled/disabled, not the individual actions.

If 2.1.0 still does no work, can you try the latest from git/master? I've made a small change that might help.

**Comment by xhpohanka** [comment]
No problem, its holiday time :)  
(unfortunately we are past deadline on our project in my job so I'm sitting at PC even now)
I will check...

**Comment by xhpohanka** [comment]
With latest git master it seems that everything works fine. Arch linux package is out of date so I switched to aur cantata-git

I also think that following feature with song priority could be useful - when multiple items are selected assign them decreasing priority, one could then enquee for example whole album in correct order. In mpd songs with same priority all played in random order.

Thanks for best mpd client available.

**Comment by CDrummond** [comment]
Thanks, I've added this 'fix' to the ChangeLog for 2.2.0

As to decreasing priority - I see the point you are making. Perhaps I could add this to the 'custom priority' dialog. e.g.
```
Priority: 250
[x] Decrease each subsequent item in the same album
```

Make sense? If so, I'll add this for 2.2.0 :-)

**Comment by xhpohanka** [comment]
> Make sense? If so, I'll add this for 2.2.0 :-)

Yes, it do. Maybe I would use 'selected items' instead of same album as it is more versatile.

**Comment by CDrummond** [comment]
Actually, that will be easier to implement :-)

**Comment by CDrummond** [comment]
Implemented for 2.2.0
